feat(ffe-tabs): Redesign av faner

kariannekristiansen opened the pull request:
Faner (tabs) har nå nye farger, fonter og former

BREAKING CHANGE: faner (tabs) har helt nytt utseende

Nytt design på faner

## Beskrivelse
**Lightmode**
Default
![Screenshot from 2022-02-11 15-14-20](https://user-images.githubusercontent.com/69858000/153608834-cf872206-7729-4dfe-bc16-274c6466d5cb.png)
Fokus
![Screenshot from 2022-02-11 15-14-07](https://user-images.githubusercontent.com/69858000/153608833-704fc911-06ab-4a26-8b66-60c141bb6b6d.png)
Hover
![Screenshot from 2022-02-11 15-26-13](https://user-images.githubusercontent.com/69858000/153610128-7c8b9c14-3a73-4472-891b-c72417481f09.png)

**Darkmode**
Defualt
![Screenshot from 2022-02-11 15-16-20](https://user-images.githubusercontent.com/69858000/153610334-eb676852-5f59-4abf-9661-0740d9234dc6.png)
Fokus
![Screenshot from 2022-02-11 15-17-13](https://user-images.githubusercontent.com/69858000/153610340-4b58dd11-e084-4606-b42e-75c9fa08c9c1.png)
Hover
![Screenshot from 2022-02-11 15-32-47](https://user-images.githubusercontent.com/69858000/153610341-db99841b-ab79-41a4-98c5-8435ebe949f6.png)


## Motivasjon og kontekst

Brukes allerede mange steder i nettbanken og nå skal team-sparing ta i bruk denne da denne tar mindre plass.

## Testing

Har testet lokalt.

<!--

Helt til slutt, har du ..

.. lest retningslinjene våre for bidrag?
.. tatt en siste sjekk for kode og skrivefeil, debug informasjon o.l?
.. kjørt og eventuelt oppdatert testene?
.. kommentert og dokumentert det som trengs?
.. knyttet denne opp til relaterte issues?

-->

kariannekristiansen committed:
feat(ffe-tabs): redesign av faner

fanene har nå nye farger, fonter og former

BREAKING CHANGE: faner har helt nytt utseende

kariannekristiansen commented:
Designeren på teamet lurer på om det er mulig med en animasjon mellom knappene. Noe lignende dette: https://codepen.io/hsynlms/pen/QZQeqq, men jeg er ikke så god på animations og har prøvd et par ting uten hell. Er dette mulig å gjøre og hvordan kan man gjøre det eventuelt?

antidecaf commented on packages/ffe-tabs/less/tab.less:
```diff
@@ -9,57 +9,84 @@
-    cursor: pointer;
-    display: block;
+    color: @ffe-farge-vann;
+    padding: 10px @ffe-spacing-lg;
```
`10px` kan med fordel erstattes med `@ffe-spacing-xs` (8px), slik at alle verdier følger [retningslinjer for spacing](https://design.sparebank1.no/profil/spacing/).

antidecaf commented on packages/ffe-tabs/less/tab.less:
```diff
@@ -9,57 +9,84 @@
+    }
+
+    &:hover:not(&--selected) {
+        color: @ffe-white;
```
`@ffe-white` er en verdi fra den gamle fargepaletten, som er deprecated og skal slettes etterhvert. Erstatt gjerne med `@ffe-farge-hvit`. :)

antidecaf commented on packages/ffe-tabs/less/tab.less:
```diff
@@ -9,57 +9,84 @@
+        border: 2px solid @ffe-farge-vann;
+        box-sizing: border-box;
+        border-radius: 22px;
+        box-shadow: 0 0 0 2px @ffe-blue-focus;
```
Denne focus-stylingen er lik stylingen fra den gamle visuelle profilen. For konsistensens skyld bør det heller løses på lignende måte som i [de oppdaterte knappene](https://design.sparebank1.no/komponenter/knapper/), med nye farger og litt annen utforming og oppførsel.

Måten det funker på der er at et pseudo-element (`.ffe-button::after`) vises på focus, men kun når man navigerer med tastatur. Relevant styling ligger i [linje 75-106](https://github.com/SpareBank1/designsystem/blob/0b3c37377454f1c1aa9ed019f11474e1df4601b6/packages/ffe-buttons/less/base-button.less#L75) i base-button.less. [Mer info om teknikken på css-tricks.com](https://css-tricks.com/the-focus-visible-trick/).

kariannekristiansen replied:
Skjønner! Takk :D

antidecaf commented on packages/ffe-tabs/less/tab.less:
```diff
@@ -9,57 +9,84 @@
 
-    &:focus,
     &:active {
+        color: @ffe-white;
```
`@ffe-white` bør erstattes med `@ffe-farge-hvit`, ref linje 32

antidecaf commented on packages/ffe-tabs/less/tab.less:
```diff
@@ -9,57 +9,84 @@
-        background-color: @ffe-white;
-        color: @ffe-blue-royal;
+        background-color: @ffe-farge-fjell;
+        color: @ffe-white;
```
`@ffe-white` bør erstattes med `@ffe-farge-hvit`, ref linje 32

pethel commented:
Vill denne ikke kunna bli problematisk på mobil? Hvordan vill den brekke? VI i sparing har ju 4 tabar noen steder med lit lengre navn. Legger dom seg under varandra?

pethel commented on packages/ffe-tabs/less/tab.less:
```diff
@@ -9,57 +9,84 @@
-        color: @ffe-blue-royal;
-        background: @ffe-white;
+    @media (max-width: @breakpoint-sm) {
+        padding: 10px 10px;
```
padding: 10px 10px; ->  padding:  @ffe-spacing-xs;

pethel commented:
> Designeren på teamet lurer på om det er mulig med en animasjon mellom knappene. Noe lignende dette: https://codepen.io/hsynlms/pen/QZQeqq, men jeg er ikke så god på animations og har prøvd et par ting uten hell. Er dette mulig å gjøre og hvordan kan man gjøre det eventuelt?

Jag tror ikke den er kjempeenkel den der hvis du skall kunna ha dynamisk antal tabs. Exemplet dit utgår ju fra att det er 2 tabs  og alltid 50%. Akkurat går vell sikkert løsa hvis man er villig og bruke tid på det. Men vad med mobil og att den muligens må brekke på flere rader osv?

pethel commented on packages/ffe-tabs/less/tab.less:
```diff
@@ -9,57 +9,84 @@
+
+    &:focus:not(&--selected) {
+        border: 2px solid @ffe-farge-vann;
+        box-sizing: border-box;
```
Denne er ikke den i noen normalize.css eller lignende allrede?

kariannekristiansen replied:
Jeg klarte ikke å finne noe :thinking:

pethel replied:
Da er det sikkert fint

pethel commented on packages/ffe-tabs/less/tab.less:
```diff
@@ -9,57 +9,84 @@
             }
         }
+
+        &:focus {
```
Det meste av dette kan vell flyttas op til blocket ovan som stylar focus/hover allrede? 
```
 &:focus {
            outline: none;
        }

        &:active {
            box-shadow: none;
            outline: none;
        }
```

kariannekristiansen commented:
> Vill denne ikke kunna bli problematisk på mobil? Hvordan vill den brekke? VI i sparing har ju 4 tabar noen steder med lit lengre navn. Legger dom seg under varandra?

Citona ville ta i bruk disse siden hun gjør en opprydding på beholdningsoversiktene. Kan høre med henne hva vi gjør der vi bruker 4 knapper, for 3 tabs er nok maks det vi kan ha. Vet at allerede kort og betaling bruker tabs i dagens løsning.

pethel commented:
> > Vill denne ikke kunna bli problematisk på mobil? Hvordan vill den brekke? VI i sparing har ju 4 tabar noen steder med lit lengre navn. Legger dom seg under varandra?
> 
> Citona ville ta i bruk disse siden hun gjør en opprydding på beholdningsoversiktene. Kan høre med henne hva vi gjør der vi bruker 4 knapper, for 3 tabs er nok maks det vi kan ha. Vet at allerede kort og betaling bruker tabs i dagens løsning.

Ja fint. Det er ju ikke kun vårt behov vi må se til her uten hele Sparebank1 utvikkling. Vi må også tenke på att det engelska oversetningarna ofta er litt lenger og tar mer plats vilket gjør kanpparna bredare.

antidecaf commented:
Som @pethel sier er denne litt problematisk på mobil i og med at layouten ikke er responsiv. Det blir plass til veldig lite innhold i bredden. Har sett litt nærmere på det sammen med @andrea-sb1 nå. Hun lager et forslag i Figma til hvordan komponenten kan se ut når den brekker over flere linjer, også kan vi se på det sammen.

Utseendemessig er det hele ganske likt tab buttons, så vi foreslår å deprecate og fase ut `TabButton`, `TabButtonGroup` og tilhørende styling i samme åndedrag som vi erstatter `Tab` og `TabGroup` med denne nye varianten. Så har vi kun den ene fanekomponenten å forholde oss til. :)

MariaOverlierBerg commented:
> > Vill denne ikke kunna bli problematisk på mobil? Hvordan vill den brekke? VI i sparing har ju 4 tabar noen steder med lit lengre navn. Legger dom seg under varandra?
> 
> Citona ville ta i bruk disse siden hun gjør en opprydding på beholdningsoversiktene. Kan høre med henne hva vi gjør der vi bruker 4 knapper, for 3 tabs er nok maks det vi kan ha. Vet at allerede kort og betaling bruker tabs i dagens løsning.

Vi på betaling har tatt i bruk opp til 3 faner og noen av de har litt lange "titler", men kan fort skje at vi vil ha mer enn det i framtida :)

andrea-sb1 commented:
> > > Vill denne ikke kunna bli problematisk på mobil? Hvordan vill den brekke? VI i sparing har ju 4 tabar noen steder med lit lengre navn. Legger dom seg under varandra?
> > 
> > 
> > Citona ville ta i bruk disse siden hun gjør en opprydding på beholdningsoversiktene. Kan høre med henne hva vi gjør der vi bruker 4 knapper, for 3 tabs er nok maks det vi kan ha. Vet at allerede kort og betaling bruker tabs i dagens løsning.
> 
> Vi på betaling har tatt i bruk opp til 3 faner og noen av de har litt lange "titler", men kan fort skje at vi vil ha mer enn det i framtida :)

Har satt opp et forslag i Figma-branchen på hvordan tab-komponenten kan se ut når den brekker over flere linjer :)

Tenker vi kan ha retningslinjer på å holde tab-tekstene så korte som mulig, akkurat som vi har for knappene, men at det kan være lurt at det finnes mulighet til å brekke dem over flere linjer de gangene det ikke er mulig å holde dem på én linje på mobil

<img width="1158" alt="Skjermbilde 1" src="https://user-images.githubusercontent.com/77671758/154061028-14b3d2b1-a917-4591-be9a-a3d934bd9d66.png">

<img width="1162" alt="Skjermbilde 2" src="https://user-images.githubusercontent.com/77671758/154061068-32458bb9-6016-41d4-a7be-d81c031f4222.png">

kariannekristiansen commented:
@antidecaf Jeg lurer på hvordan det er best å løse breakpoints. Vil jo helst at det skal se slik ut 
![image](https://user-images.githubusercontent.com/69858000/154672377-768be680-5f2a-4eca-a4ae-de4d9dc20443.png)
 når det er plass på mobil og knekke når labels blir for lange eller det er flere enn 4 Tabs. Har du forslag til hvordan man kan løse det best mulig? Kan sette breakpoint på skjermstørrelse, men da blir det slik som TabGroupButton er i dag. :thinking:

antidecaf commented:
@kariannekristiansen Et alternativ kan kanskje være å legge til en prop/modifier som overstyrer styling slik at layouten ikke brekker over flere linjer, så kan den evt tas i bruk når man vet at innholdet får plass. Men jeg er ikke helt sikker på hvordan det best kan løses "automatisk".

kariannekristiansen commented:
> @kariannekristiansen Et alternativ kan kanskje være å legge til en prop/modifier som overstyrer styling slik at layouten ikke brekker over flere linjer, så kan den evt tas i bruk når man vet at innholdet får plass. Men jeg er ikke helt sikker på hvordan det best kan løses "automatisk".

Ja, har tenkt på det.

pethel commented on packages/ffe-tabs-react/src/TabGroup.js:
```diff
@@ -2,12 +2,12 @@ import React from 'react';
             className={classNames(
                 'ffe-tab-group',
-                { 'ffe-tab-group--thin': thin },
+                { 'ffe-tab-group--noBreak': noBreak },
```
'ffe-tab-group--no-break' for og vare konsekvent

pethel commented on packages/ffe-tabs/less/tab-group.less:
```diff
@@ -12,42 +12,61 @@
-    &--thin {
-        margin: 0;
+    &--noBreak {
+        display: flex;
```
er den ikke flex allrede?

kariannekristiansen committed:
feat(ffe-tabs): redesign av faner

fanene har nå nye farger, fonter og former

BREAKING CHANGE: faner har helt nytt utseende

pethel reviewed (commented):
@kariannekristiansen husk og fikse TS typerne

pethel commented:
@kariannekristiansen så burde vell commitesen fordeles på ffe-tabs og ffe-tabs-react. Både er vell breaking changes
